tests/model_registry/utils.py

```diff
@@ -156,3 +161,71 @@ def get_model_registry_db_label_dict(db_resource_name: str) -> dict[str, str]:
         Annotations.KubernetesIo.INSTANCE: db_resource_name,
         Annotations.KubernetesIo.PART_OF: db_resource_name,
     }
+
+
+def get_pod_container_error_status(pod: Pod) -> str:
+    pod_instance_status = pod.instance.status
+    # Check the containerStatuses and if any containers is in waiting state, return that information:
+
+    for container_status in pod_instance_status.get("containerStatuses", []):
+        if waiting_container := container_status.get("state", {}).get("waiting"):
+            return waiting_container["reason"] if waiting_container.get("reason") else waiting_container
+        return ""
+    return ""
+
+
+def get_not_running_pods(pods: list[Pod]) -> list[Any]:
```

Comment:
why can't `utilities.infra.verify_no_failed_pods` be used?

Reply:
Use case does not seem relevant. I don't need to look at pods by labels etc.